src/store.js

```diff
@@ -1,15 +1,23 @@
 import { useMemo } from 'react';
 import { observable } from '@nx-js/observer-util';
 
-import { isInsideFunctionComponent } from './view';
+import {
+  isInsideFunctionComponent,
+  isInsideStatelessComp,
+} from './view';
 import { hasHooks } from './utils';
 
 export function store(obj) {
   // do not create new versions of the store on every render
   // if it is a local store in a function component
   // create a memoized store at the first call instead
-  if (hasHooks && isInsideFunctionComponent) {
+  if (isInsideFunctionComponent) {
```

Comment:
I think there is no need to change the `if` logic here. Removing `hasHooks` from the block will cause easy-state to wrap the passed object with an observable BEFORE it throws an error which is an undesired side effect.

I think it should be:

```js
if (hasHooks && isInsideFunctionComponent) { /* wrap with observable */}
if (!hasHooks && isInsideFunctionComponent) { /* throw an error */ }
```

I also don't see the need for the new `isInsideStatelessComp`. It seems to be an exact duplicate of `isInsideFunctionComponent`. Am I missing something?

Reply:
We had to change the logic since `isInsideFunctionComponent` will only be true if `hasHooks` is true, there is simply no way for `(!hasHooks && isInsideFunctionComponent)` to be true. Check the `view.js` file at https://github.com/RisingStack/react-easy-state/blob/7ac5afb7b86ac3ec18fece50d3818f4eb777a6a9/src/view.js#L43

Reply:
Oh, true. Still `isInsideFunctionComponent` is set to true only while the render is running of a function component. `isInsideStatelessComponent` is set to true when the render starts and it is left like that afterwards. I think this is a bit fishy and needs to be improved a bit.

To clarify what needs to be done: I think we have to throw the error when a `store` is called in [here](https://github.com/RisingStack/react-easy-state/blob/master/src/view.js#L97).

Reply:
What do you think about addig an `isInsideClassComponent` flag?

Also to code can run into there when hooks are not supported and a function is wrapped with `view` and when the user wraps a class with `view`. In both cases a `store` should not be created inside the render but it raises the question if we should throw a different error in case #2 (when `store` is called in a class render by the user).